storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java

```diff
@@ -318,6 +318,11 @@ public WeakSubjectivityState getWeakSubjectivityState() {
     return finalizedDao.getFinalizedBlockAtSlot(slot);
   }
 
+  @Override
+  public Optional<UInt64> getEarliestHistoricalBlockSlot() {
```

Comment:
I thought it might be confusing to have an API named `getEarliestFinalizedBlockSlot` because it sounds like there's an earliest finalized block on chain, when really we're just want to know the earliest available block.  Probably a better name would be `getEarliestAvailableFinalizedBlockSlot` - will update.